### ISSUE TITLE
urldownload API endpoint?

### ISSUE THREAD

**Issue by xxdesmus** [viper-framework/viper#679]
I might just be missing something obvious, but it doesn't look like there's an API endpoint for uploading data via a URL, only via a local file. In the web UI there's a call to urldownload -- but I don't see anything in the document or code about a corresponding API endpoint.  Any tips?

**Comment by frennkie** [comment]
No, you are right.. There is no API endpoint for performing a Viper file upload from an URL.
Would you be willing to work on a PR for this?
I'd happy to check/review it. 

On 5 July 2018 21:53:16 CEST, Justin <notifications@github.com> wrote:
>I might just be missing something obvious, but it doesn't look like
>there's an API endpoint for uploading data via a URL, only via a local
>file. In the web UI there's a call to urldownload -- but I don't see
>anything in the document or code about a corresponding API endpoint. 
>Any tips?
>
>-- 
>You are receiving this because you are subscribed to this thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/viper-framework/viper/issues/679

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by frennkie** [comment]
Although I didn't initially think this is needed.. When you are calling the API endpoint you are working programmatically already... So you could simply download the file before calling the API in your client...

**Comment by xxdesmus** [comment]
You'd be terrified if you saw any code I worked out. Unfortunately I'm nowhere near competent enough to do the PR. 

In terms of use case: yeah, I'm just attempting to avoid downloading every file just to then save it into Viper to then delete the original file -- vs. just pulling it from the URL directly into Viper.

**Comment by frennkie** [comment]
@xxdesmus Feel free to work on this - I'll be happy to review it. If you need any pointers let me know.
